File: workspace/2023/2023-06-18/numerical-sequence.prolog/fibonacci.pl

fibonacci(0, 0).

fibonacci(1, 1).

fibonacci(N, Fn) :-
    N > 1,
    is(A, N - 2),
    is(B, N - 1),
    fibonacci(A, Fa),
    fibonacci(B, Fb),
    is(Fn, Fa + Fb).

File: workspace/2025/2025-10-25/numerical-sequence.forth/fibonacci.fs

: _fibonacci recursive ( a b n -- fn )
    dup 0 = if
        drop drop
    else
        rot rot tuck + rot ( a b n -- b a+b n )
        1 - recurse
    endif
;

: fibonacci ( n -- fn )
    0 1 rot _fibonacci
;

: print_fibonacci_numbers_range recursive ( max min -- )
    over over < if
        drop drop
    else
        dup fibonacci .
        1 + recurse
    endif
;

: print_fibonacci_numbers ( n -- )
    s\" fibonacci numbers:\n" type
    0 print_fibonacci_numbers_range
    s\" \n" type
;

